packages/filesystem/src/node/node-filesystem.ts

```diff
@@ -356,6 +353,40 @@ export class FileSystemNode implements FileSystem {
         return children;
     }
 
+    /** Return true if it's possible for this URI to have children.
+     *  It might not be possible to be certain because of permission problems
+     *  Or other filesystem errors.
+     */
+    protected mayHaveChildren(uri: URI): boolean {
+        /* If there's a problem reading the root directory.
+           Assume it's not empty to avoid overwriting anything.  */
+        try {
+            const rootStat = this.doGetStat(uri, 0);
+            if (rootStat === undefined) {
+                return true;
+            }
+            /* Not a directory.  */
+            if (rootStat !== undefined && rootStat.isDirectory === false) {
+                return false;
+            }
+        } catch {
+            return true;
```

Comment:
Could it be that we would swallow real issues here or they would pop up somewhere anyway?

Reply:
They would pop up anyway in move or another "real" operation.